pkg/web/router.go

```diff
@@ -132,6 +137,16 @@ func NewRouter(config *RouterConfig, wpool pool.WorkerPoolInterface) http.Handle
 		r.Use(authorizationMiddleware)
 	}).(*chi.Mux)
 
+	var oauth2Context *authentication.OAuth2Context
+
+	if oauth2Config.Enabled {
+		oauth2Context = authentication.NewOAuth2Context(config.oauth2, oidc.NewProvider, tracer, logger, monitor)
+
+		authenticationMiddleware := authentication.NewAuthenticationMiddleware(oauth2Context, tracer, logger)
+		authenticationMiddleware.SetAllowListedEndpoints("/api/v0/login", "/api/v0/status", "api/v0/metrics")
+		apiRouter = apiRouter.With(authenticationMiddleware.OAuth2Authentication).(*chi.Mux)
```

Comment:
I switched the other routers to use `Use`, should we change this one as well? (the issue that we had with `Use` does not happen with subrouters)